lib/redis/objects/connection_pool_proxy.rb

```diff
@@ -6,8 +6,8 @@ def initialize(pool)
         @pool = pool
       end
 
-      def method_missing(name, *args, &block)
-        @pool.with { |x| x.send(name, *args, &block) }
+      def method_missing(name, *args, **kargs, &block)
+        @pool.with { |x| x.send(name, *args, **kargs, &block) }
```

Comment:
Unfortunately that's not the proper fix. 

You want to use `ruby2_keywords` here. See: https://eregon.me/blog/2021/02/13/correct-delegation-in-ruby-2-27-3.html

Reply:
This fix anyway pass 4 arguments to zrangebyscore instead 3, last element of args is empty hash.

https://github.com/redis/redis-rb/blob/fa76a2661739f56e3458fc5ed5a7b7c31861eff5/lib/redis.rb#L1963

Reply:
Can somebody provide a proper fix that is Ruby 3 compatible?  I'm not using Ruby 3 yet in production and don't typically use connection pooling.  Would appreciate anyone who can apply the blog that @byroot linked.

```
def foo(*args, &block)
  target(*args, &block)
end
ruby2_keywords :foo if respond_to?(:ruby2_keywords, true)
```

Reply:
Here you go https://github.com/nateware/redis-objects/pull/261